src/client.cpp

```diff
@@ -849,8 +907,20 @@ void CClient::OnControllerInMuteMyself ( bool bMute )
     emit ControllerInMuteMyself ( bMute );
 }
 
-void CClient::OnClientIDReceived ( int iChanID )
+void CClient::OnClientIDReceived ( int iServerChanID )
 {
+    // if we have just connected to a running server, iActiveChannels will be 0
+    // if iActiveChannels is not 0, the server must have been restarted on the fly
+    // in that case, channels might have changed, so clear our list to get it afresh.
```

Comment:
Does this new code prevent the old trashing of "No fader sort" by this event?  Given the server channel id isn't used, the "no sort order" should remain stable, too?

Reply:
I don't think it will be guaranteed to be any better. It will depend on the order that the other clients appear back on the server, and thereby appear in a client list sent to us by that server. If we receive them in a different order from what they were, they might get recreated in a different order.

I don't think it will be any worse though.